<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Compute Bayesian credible interval for mediating effect</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mediation</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Compute Bayesian credible interval for mediating effect</h1>

</div>

<div id="TOC">
<ul>
<li><a href="#data-and-packages">Data and packages</a></li>
<li><a href="#heart-vs.kidney-data">Heart vs. Kidney data</a></li>
<li><a href="#conclusions">Conclusions</a></li>
<li><a href="#session-information">Session information</a></li>
</ul>
</div>

<!-- The file analysis/chunks.R contains chunks that define default settings
shared across the workflowr files. -->
<!-- Update knitr chunk options -->
<!-- Insert the date the file was last updated -->
<p><strong>Last updated:</strong> 2018-03-30</p>
<!-- Insert the code version (Git commit SHA1) if Git repository exists and R
 package git2r is installed -->
<p><strong>Code version:</strong> b29d229</p>
<hr />
<div id="data-and-packages" class="section level2">
<h2>Data and packages</h2>
<pre class="r"><code>library(medinome)
library(limma)
library(qvalue)
library(MCMCpack)</code></pre>
<hr />
</div>
<div id="heart-vs.kidney-data" class="section level2">
<h2>Heart vs. Kidney data</h2>
<p>Take from Lauren’s data comparing kidney versus heart tissue samples.</p>
<pre class="r"><code>df &lt;- get(load(&quot;../data/example-kidney-v-heart-human.rda&quot;))
Y=df$exprs_pair
X=df$tissue
M=df$methyl_pair

# scale gene expression to mean 0 and standard deviation 1 for each gene
Y &lt;- t(scale(t(Y)))
M &lt;- t(scale(t(M)))

# apply limma to scaled expression to select DE genes
design_1 &lt;- model.matrix(~X)
Y_counts &lt;- 2^Y
Y_voom &lt;- voom(Y_counts, design=design_1, normalize.method = &quot;none&quot;)
model_1 &lt;- lmFit(Y_voom, design_1)
model_1 &lt;- eBayes(model_1)

# qvaule package for FDR control
qval &lt;- qvalue(model_1$p.value[,2])

# select DE genes with q-value  &lt; .05
ii.de &lt;- which(qval$qvalues &lt; .05)
Y.de &lt;- Y[ii.de, ]
M.de &lt;- M[ii.de, ]

# create a permutated dataset
M.permute &lt;- do.call(rbind, lapply(1:nrow(M.de), function(g) {
  m.perm &lt;- sample(M.de[g,])
  return(m.perm)
  }))

# fitting mediation test
fit.voom &lt;- mediate.test.voom(Y=Y.de, X=X, M=M.de)

fit.voom.perm &lt;- mediate.test.voom(Y=Y.de, X=X, M=M.permute)</code></pre>
<p>Function to compute Bayesian credible interval.</p>
<pre class="r"><code>ci &lt;- vector(&quot;list&quot;, nrow(Y.de))
type_1 &lt;- 0.05
prior_mn_model_3 &lt;- rep(0,3)
prior_mn_model_2 &lt;- rep(0,2)

for (g in 1:nrow(Y.de)) {

  prior_sd_model_2 &lt;- c(fit.voom$sigma_gamma2[g], fit.voom$sigma_alpha[g])
  prior_sd_model_3 &lt;- c(fit.voom$sigma_gamma3[g], fit.voom$sigma_tau_prime[g],
                        fit.voom$sigma_beta[g])
  
  M_g &lt;- M.de[g,]
  Y_g &lt;- Y.de[g,]

  model_2 &lt;- MCMCregress(M_g ~ X, burnin = 1000 , mcmc = 1000, 
              seed = 7, b0= prior_mn_model_2, B0 = prior_sd_model_2^(-1))
              # sigma.mu = fit.voom$sigma_model2[g]^2,
              # sigma.var = var(fit.voom$sigma_model2^2))
  
  model_3 &lt;- MCMCregress(Y_g ~ X + M_g, burnin = 1000 , mcmc = 1000, 
              seed = 7, b0= prior_mn_model_3, B0 = prior_sd_model_3^(-1))
              # sigma.mu = fit.voom$sigma_model3[g]^2,
              # sigma.var = var(fit.voom$sigma_model3^2))

  ab &lt;- model_2[,2]*model_3[,3]

  ci.lower &lt;- quantile(ab,probs=type_1/2,type=4)
  ci.higher &lt;- quantile(ab,probs=(1-type_1/2),type=4)

  ci[[g]] &lt;- data.frame(ci.lower, ci.higher)
}
ci &lt;- do.call(rbind, ci)

res &lt;- sapply(1:g, function(g) {fit.voom$ab[g] &lt; ci[g,1] | fit.voom$ab[g] &gt; ci[g,2]})
mean(res)</code></pre>
<pre><code>[1] 0.03683763</code></pre>
<pre class="r"><code>sum(res)</code></pre>
<pre><code>[1] 130</code></pre>
<pre class="r"><code>ci.perm &lt;- vector(&quot;list&quot;, nrow(Y.de))
type_1 &lt;- 0.05
prior_mn_model_3 &lt;- rep(0,3)
prior_mn_model_2 &lt;- rep(0,2)

for (g in 1:nrow(Y.de)) {

  prior_sd_model_2 &lt;- c(fit.voom.perm$sigma_gamma2[g], fit.voom.perm$sigma_alpha[g])
  prior_sd_model_3 &lt;- c(fit.voom.perm$sigma_gamma3[g], 
                        fit.voom.perm$sigma_tau_prime[g],
                        fit.voom.perm$sigma_beta[g])
  
  M_g &lt;- M.permute[g,]
  Y_g &lt;- Y.de[g,]

  model_2 &lt;- MCMCregress(M_g ~ X, burnin = 1000 , mcmc = 1000, 
              seed = 7, b0= prior_mn_model_2, B0 = prior_sd_model_2^(-1))
              # sigma.mu = fit.voom.perm$sigma_model2^2,
              # sigma.var = var(fit.voom.perm$sigma_model2^2))
  
  model_3 &lt;- MCMCregress(Y_g ~ X + M_g, burnin = 1000 , mcmc = 1000, 
              seed = 7, b0= prior_mn_model_3, B0 = prior_sd_model_3^(-1))
              # sigma.mu = fit.voom.perm$sigma_model3^2,
              # sigma.var = var(fit.voom.perm$sigma_model3^2))

  ab &lt;- model_2[,2]*model_3[,3]

  ci.lower &lt;- quantile(ab,probs=type_1/2,type=4)
  ci.higher &lt;- quantile(ab,probs=(1-type_1/2),type=4)

  ci.perm[[g]] &lt;- data.frame(ci.lower, ci.higher)
}
ci.perm &lt;- do.call(rbind, ci.perm)

res.perm &lt;- sapply(1:g, function(g) {fit.voom.perm$ab[g] &lt; ci.perm[g,1] | fit.voom.perm$ab[g] &gt; ci.perm[g,2]})
mean(res.perm)</code></pre>
<pre><code>[1] 0.01558515</code></pre>
</div>
<div id="conclusions" class="section level2">
<h2>Conclusions</h2>
<ul>
<li>Preliminary results show that only 3 percent of the genes, there’s significant effect of mediation. Using permuted methylation data, we found significant effect in only 1 percent of the genes. Though this trend is consistent with the expectation, I have little confidence in this approach given the low percentage of genes found signficant in the test of mediation coupled with the potentiality for confounding between the relations between methylation and expression.</li>
</ul>
<hr />
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ltm_1.1-0        polycor_0.7-9    msm_1.6.6        assertthat_0.2.0
 [5] MCMCpack_1.4-2   MASS_7.3-49      coda_0.19-1      qvalue_2.10.0   
 [9] limma_3.34.9     medinome_0.0.1   ggplot2_2.2.1   

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.15       pillar_1.2.1       compiler_3.4.1    
 [4] git2r_0.21.0       plyr_1.8.4         tools_3.4.1       
 [7] digest_0.6.15      evaluate_0.10.1    tibble_1.4.2      
[10] gtable_0.2.0       lattice_0.20-35    rlang_0.2.0       
[13] Matrix_1.2-12      yaml_2.1.18        expm_0.999-2      
[16] mvtnorm_1.0-7      SparseM_1.77       stringr_1.3.0     
[19] knitr_1.20         MatrixModels_0.4-1 rprojroot_1.3-2   
[22] grid_3.4.1         survival_2.41-3    rmarkdown_1.9     
[25] reshape2_1.4.3     magrittr_1.5       backports_1.1.2   
[28] scales_0.5.0       htmltools_0.3.6    mcmc_0.9-5        
[31] splines_3.4.1      colorspace_1.3-2   quantreg_5.35     
[34] stringi_1.1.6      lazyeval_0.2.1     munsell_0.4.3     </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
